go-controller/pkg/ovn/policy.go

```diff
@@ -321,8 +344,7 @@ func (oc *Controller) syncNetworkPolicies(networkPolicies []interface{}) error {
 	// remove stale egress and ingress allow arp ACLs that were leftover as a result
 	// of ACL migration for "ARPallowPolicy" when the match changed from "arp" to "(arp || nd)"
 	p = func(item *nbdb.ACL) bool {
-		return strings.Contains(item.Match, " && arp") &&
-			strings.Contains(*item.Name, arpAllowPolicySuffix)
+		return strings.Contains(item.Match, " && "+staleArpAllowPolicyMatch)
```

Comment:
Since we are not matching by suffix, would we fill more comfortable if we additionally matched by the external id? Assuming the external id should be there....

Reply:
Adding externalID here doesn't help us much since the same externalID is there for all default ingress ACLs. I thought since the one thing we are certain about is the `arp` match being wrong, let's go only by that. Do we have any value in adding the externalID here?

Reply:
the same externalID is there for all default deny ACLs but not all ACLs. So we would be matching "arp" only within default deny ACLs and not all ACLs. It has the same use as the one we are giving it in this other predicate we use in line 215
```
		return item.ExternalIDs[defaultDenyPolicyTypeACLExtIdKey] == string(npType) && // default-deny-policy-type:Egress or default-deny-policy-type:Ingress
			strings.Contains(item.Match, gressSuffix) && // Match:inport ==	@ablah80448_egressDefaultDeny or Match:inport == @ablah80448_ingressDefaultDeny
			!strings.Contains(*item.Name, arpAllowPolicySuffix) && // != name: namespace_ARPallowPolicy
			!strings.Contains(*item.Name, gressSuffix) // filter out already converted ACLs (we still continue to pick the cases where names were >45 chars)
			!strings.Contains(item.Match, staleArpAllowPolicyMatch) && // != match: (arp || nd) OR arp
```

Reply:
yeah better safe than sorry, I will use externalIDs